README.md

```diff
@@ -1,2 +1,164 @@
+
+Specifications that conform to the contract are, with the help of reference databases, able to unambiguously cross-reference each other's definitions.
+
+## Some examples
+
+A simple example of a definition:
+
+```HTML
+<dfn id="simple-def" data-export="" data-dfn-type="dfn">
+  I'm a simple definition
+</dfn>
+```
+
+A more complex examples:
+
+```HTML
+<!-- Example of defining an abstract operation -->
+<dfn
+  id="string-parser"
+  data-export=""
+  data-lt="parse a string"
+  data-dfn-type="abstract-op">
+  string parser
+</dfn>
+
+<!-- Example of defining a WebIDL interface -->
+<dfn
+  data-export=""
+  data-dfn-type="interface"
+  id="dom-paymentrequest"
+  data-dfn-for="">
+  <code>PaymentRequest</code>
+</dfn>
+
+```
+
+## Conformance
+
+This document serves as the canonical source for how definitions are marked-up.
+
+Conforming applications to this specification are primarily authoring tools that assist in generating technical specifications. However, this specification is also here to help people who wish to mark up documents manually (or simply want to understand why various attributes mean and what they do!).
+
+Example of authoring tools (or "generators") that try to conform to this specification:
+
+- [Bikeshed](http://github.com/tabatkins/bikeshed)
+- [ReSpec](http://github.com/w3c/respec)
+- [Wattsi](https://github.com/whatwg/wattsi)
+
+Note: An accompanying test suite helps assure authoring tools conform to the contract.
+
+## Who consumes these definitions
+
+Aside from specifications internally cross-referencing their own terms and terms in other specifications, "crawlers" extract terms defined in specifications into structured data.
+
+Examples of definition crawlers include:
+
+- [Shepherd](https://dev.csswg.org/projects/shepherd)
+- [Reffy](https://github.com/w3c/reffy)
+
+Crawlers structure and categorized extracted terms into files or databases, allowing for the creation of search tools such as:
+
+- <https://respec.org/xref>
+- Bikeshed's cross-reference search.
+
+## How to get your spec indexed
+
+To have your specification indexed by a crawler, it MUST be registered with:
+
+- [browser specs](https://github.com/w3c/browser-specs/) - See [criteria for inclusion](https://github.com/w3c/browser-specs/blob/master/README.md#spec-selection-criteria).
+- [Shepherd](https://dev.csswg.org/projects/shepherd) - [Contact maintainer(s) directly](https://dev.csswg.org/users/3).
+
+## The Contract
+
+The following sections define the elements, attributes, and attribute values (along with any specific micro-syntax) that constitute the contract.
+
+### Definitions
+
+A <dfn>definition</dfn> is:
+
+- a `<dfn>` element.
+- a `h1`-`h6` element.
+
+No other HTML elements are recognized as defining a term.
+
+#### dfn
+A a dfn-definition MUST have the following attributes:
+
+- `id` attribute, unique to the document.
+- `data-dfn-type`, with one of type values.
```

Comment:
I think we should recognize it as optional for `<dfn>` elements (that's how reffy handles it, and I suspect shepherd too)

Reply:
I was going by [4 of the contract](https://tabatkins.github.io/bikeshed/#dfn-contract):

>  `data-dfn-type` MUST be provided, and set one of the accepted values.

Reply:
I don't have a strong preference tho. ReSpec treats them as optional, but I sent a PR to ReSpec to always include them: https://github.com/w3c/respec/pull/3667.

Reply:
It is a reasonable default especially if we consider that people could possibly create spec markup manually (no one really does that, but the intro suggests that is a possibility): `<dfn data-dfn-type="dfn">` looks very redundant when compared to `<dfn>`.

Reply:
+1 on it being super-redundant. And there *are* manually maintained specs, e.g. WebGL 1 & 2.

Reply:
Shepherd does some inference if it's missing, for legacy specs. I don't think we want to specify those inference rules; they're heuristic and kinda complex and specific to the styles that happen to be used by the legacy specs in question. (They're based on examining the text, the ID, and I think there's some matching to WebIDL in the document as well.)

So I def prefer continuing to require it.